tests/dryrun/install_test.go

```diff
@@ -0,0 +1,359 @@
+package dryrun
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	k0sv1beta1 "github.com/k0sproject/k0s/pkg/apis/k0s/v1beta1"
+	"github.com/replicatedhq/embedded-cluster/pkg/defaults"
+	dryruntypes "github.com/replicatedhq/embedded-cluster/pkg/dryrun/types"
+	"github.com/replicatedhq/embedded-cluster/pkg/kubeutils"
+	"github.com/replicatedhq/embedded-cluster/pkg/release"
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/yaml"
+)
+
+func TestDefaultInstallation(t *testing.T) {
+	releaseData, err := os.ReadFile("./assets/install-release.yaml")
+	if err != nil {
+		t.Fatalf("fail to read release data: %v", err)
+	}
+	if err := release.SetReleaseDataForTests(map[string][]byte{
+		"release.yaml": releaseData,
+	}); err != nil {
+		t.Fatalf("fail to set release data: %v", err)
+	}
+
+	// file to store the dryrun output
+	drFile := filepath.Join(t.TempDir(), "ec-dryrun.yaml")
+
+	if err := runEmbeddedClusterCmd(
+		"install",
+		"--dry-run", drFile,
+		"--no-prompt",
+		"--license", "./assets/install-license.yaml",
+	); err != nil {
+		t.Fatalf("fail to dryrun install embedded-cluster: %v", err)
+	}
+
+	stdout, err := exec.Command("cat", drFile).Output()
+	if err != nil {
+		t.Fatalf("fail to get dryrun output: %v", err)
+	}
+
+	dr := dryruntypes.DryRun{}
+	if err := yaml.Unmarshal([]byte(stdout), &dr); err != nil {
+		t.Fatalf("fail to unmarshal dryrun output: %v", err)
+	}
```

Comment:
you could probably make convenience functions in the dryrun package for this

Reply:
true